File: Backend/hse-service/src/main/java/tn/esprit/hseservice/service/InspectionService.java
package tn.esprit.hseservice.service;

import tn.esprit.hseservice.entity.Inspection;
import java.util.List;

public interface InspectionService {
    Inspection createInspection(Inspection inspection);
    Inspection updateInspection(Long id, Inspection inspection);
    void deleteInspection(Long id);
    Inspection getInspectionById(Long id);
    List<Inspection> getAllInspections();
}

File: Frontend/FrontOffice_ConstructionIQ/src/app/services/inspection.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Inspection } from 'src/models/inspection.model';

@Injectable({
  providedIn: 'root'
})
export class InspectionService {
  private baseUrl = 'http://localhost:8081/api/inspections';

  constructor(private http: HttpClient) { }

  createInspection(inspection: Inspection): Observable<Inspection> {
    return this.http.post<Inspection>(this.baseUrl, inspection);
  }

  updateInspection(id: number, inspection: Inspection): Observable<Inspection> {
    return this.http.put<Inspection>(`${this.baseUrl}/${id}`, inspection);
  }

  deleteInspection(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }

  getInspectionById(id: number): Observable<Inspection> {
    return this.http.get<Inspection>(`${this.baseUrl}/${id}`);
  }

  getAllInspections(): Observable<Inspection[]> {
    return this.http.get<Inspection[]>(this.baseUrl);
  }
}
